include/lbann/layers/io/target/target_layer.hpp

```diff
@@ -73,23 +73,38 @@ class target_layer : public generic_target_layer {
   inline void initialize_io_buffer(lbann_comm *comm, int num_parallel_readers, std::map<execution_mode, generic_data_reader *> data_readers) {
     generic_target_layer::initialize_io_buffer<T_io_buffer>(comm, num_parallel_readers, data_readers);
   }
-  
+
   data_layout get_data_layout() const override { return T_layout; }
+  El::Device get_device_allocation() const override { return Dev; }
 };
 
 template<>
-inline void target_layer<partitioned_io_buffer, data_layout::MODEL_PARALLEL>::validate_data_layout() {
+inline void target_layer<partitioned_io_buffer, data_layout::MODEL_PARALLEL, El::Device::CPU>::validate_data_layout() {
+  static_assert(true, "target_layer with partitioned_io_buffer does not supports MODEL_PARALLEL data layout");
```

Comment:
This doesn't do anything. See my comment in input_layer.hpp. Same problem as <partitioned_io_buffer, MODEL_PARALLEL, GPU> template specification.

Reply:
fixed